app/controllers/sessions_controller.rb

```diff
@@ -1,12 +1,14 @@
 class SessionsController < ApplicationController
+  before_action :find_user_by_email, only: :create
+
   def new; end
 
   def create
-    user = User.find_by(email: params[:session][:email].downcase)
-    if user&.authenticate(params[:session][:password])
+    if @user&.authenticate params[:session][:password]
```

Comment:
```suggestion
    if @user&.authenticate params.dig(:session, :password)
```
Dùng dig e nha
Tìm hiểu cách truy cập dig khác gì với truy cập trực tiếp thông thường qua []

Reply:
### [Dig method](https://docs.ruby-lang.org/en/3.0/dig_methods_rdoc.html)
Ví dụ:
```ruby
item = {
  id: "0001",
  type: "donut",
  name: "Cake",
  ppu: 0.55,
  batters: {
    batter: [
      {id: "1001", type: "Regular"},
      {id: "1002", type: "Chocolate"},
      {id: "1003", type: "Blueberry"},
      {id: "1004", type: "Devil's Food"}
    ]
  }
}
```
- Truy cập sử dụng toán tử []: 
   - `item[:batters][:batter][1][:type] # => Trả về "Chocolate"`
   - `item[:topping] # => raises NoMethodError (undefined method [] for nil:NilClass)`
- Sử dụng dig:
   - `item.dig(:batters, :batter, 1, :type) # => "Chocolate"`
   - `item[:topping] # => nil`

Như vậy dig là một phuong thức giúp truy cập và trả về giá trị của phần tử cho **nested data structures**. Khác với `[]`, dig sẽ trả về nil (kết hợp với if/elsif) nếu không thấy phần tử có key tương ứng thay vì là exception như `[]`.